File: common/src/main/java/net/impleri/playerskills/network/Manager.java
package net.impleri.playerskills.network;

import dev.architectury.networking.simple.MessageType;
import dev.architectury.networking.simple.SimpleNetworkManager;
import net.impleri.playerskills.PlayerSkills;

public abstract class Manager {
    public static SimpleNetworkManager NET = SimpleNetworkManager.create(PlayerSkills.MOD_ID);

    public static MessageType SYNC_SKILLS = NET.registerS2C("sync_skills", SyncSkillsMessage::new);

    public static MessageType RESYNC_SKILLS = NET.registerC2S("resync_skills", ResyncSkillsMessage::new);

    public static void register() {
        PlayerSkills.LOGGER.debug("Registered network channels");
    }
}

File: common/src/main/kotlin/net/impleri/playerskills/network/Manager.kt
package net.impleri.playerskills.network

import dev.architectury.networking.simple.MessageType
import dev.architectury.networking.simple.SimpleNetworkManager
import net.impleri.playerskills.PlayerSkills
import net.impleri.playerskills.utils.PlayerSkillsLogger

object Manager {
  private val NET: SimpleNetworkManager = SimpleNetworkManager.create(PlayerSkills.MOD_ID)

  val SYNC_SKILLS: MessageType = NET.registerS2C("sync_skills") { SyncSkillsMessage(it) }

  val RESYNC_SKILLS: MessageType = NET.registerC2S("resync_skills") { ResyncSkillsMessage(it) }

  fun register() {
    PlayerSkillsLogger.SKILLS.debug("Registered network channels")
  }
}
